### PR TITLE
add updateScript to various packages

### PR DESCRIPTION
**Pull request by felschr**
###### Description of changes
Added `passthru.updateScript` for `tor-browser-bundle-bin`, `mullvad-browser`, `ldtk`, `pixelorama`, `etebase-server` & `filebot`. Except for `tor-browser-bundle-bin`, I'm the maintainer of all these packages.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by felschr** [comment]
Updated `updateScript` for `tor-browser-bundle-bin` & `mullvad-browser` to use published sha256 hashes and verify their signatures similar to `firefox-bin`'s update script.

**Comment by felschr** [comment]
Result of `nixpkgs-review pr 231747` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by drupol — APPROVED** [review]
LGTM

File: pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix [hunk 1]
```diff
@@ -447,6 +456,14 @@ stdenv.mkDerivation rec {
     runHook postInstall
   '';
 
+  passthru = {
+    inherit sources;
+    updateScript = import ./update.nix {
+      inherit lib writeShellScript coreutils gnused gnugrep curl gnupg nix common-updater-scripts;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:462]: Use callPackage then we do not need to list the individual packages

File: pkgs/applications/networking/browsers/tor-browser-bundle-bin/update.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+# suffix used to match published archive
+, suffix ? "_ALL.tar.xz"
+}:
+
+writeShellScript "update-${pname}" ''
+  PATH="${lib.makeBinPath [ coreutils curl gnugrep gnused gnupg nix common-updater-scripts ]}"
+  set -euo pipefail
+
+  url=${baseUrl}
+  version=`curl -s $url \
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/tor-browser-bundle-bin/update.nix:27]:
```suggestion
  version=$(curl -s $url \
```

File: pkgs/applications/networking/browsers/tor-browser-bundle-bin/update.nix [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            | sed -rne 's,^.*href="([0-9]+(\.[0-9]+)*)/".*,\1,p' \
+            | sort --version-sort | tail -1`
+
+  if [[ "${version}" = "$version" ]]; then
+      echo "The new version same as the old version."
+      exit 0
+  fi
+
+  HOME=`mktemp -d`
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/browsers/tor-browser-bundle-bin/update.nix:36]:
```suggestion
  HOME=$(mktemp -d)
```
And this is missing a trap to clean this up

**Review by SuperSandro2000 — COMMENTED** [review]
The git updater can probably just be replaced by nix-update, then we don't need to change anything.

**Comment by felschr** [comment]
Switched all the `gitUpdater`s to `nix-update-script`, updated the Tor Browser update script and checked that they all still work.

**Comment by felschr** [comment]
Rebased the changes because of a conflict with the last `mullvad-browser` update.

File: pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix [hunk 4]
```diff
@@ -51,6 +51,16 @@
...
 , disableContentSandbox ? false
 
+# Updater
+, writeShellScript
+, curl
+, gnugrep
+, gnused
+, gnupg
+, nix
+, common-updater-scripts
```
Comment (panicgh) [inline pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix:61]: Maybe I overlook something, but now that you use `callPackage`, are these still needed?

```suggestion
```

File: pkgs/applications/networking/browsers/mullvad-browser/default.nix [hunk 5]
```diff
@@ -42,6 +42,17 @@
...
 
+# Updater
+, writeShellScript
+, coreutils
+, curl
+, gnugrep
+, gnused
+, gnupg
+, nix
+, common-updater-scripts
```
Comment (panicgh) [inline pkgs/applications/networking/browsers/mullvad-browser/default.nix:53]: Maybe I overlook something, but now that you use `callPackage`, are these still needed?

```suggestion
```

File: pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix (hunk 4 above)
Comment (felschr) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix]: Of course, you're absolutely right. I didn't really understand that part about `callPackage`, even though I've used it like that before 😅.

Comment (felschr) [reply pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix]: Btw, did you see my comment here? https://github.com/NixOS/nixpkgs/pull/231708#issuecomment-1546856364

**Comment by felschr** [comment]
Can we merge this? I'll be gone for a week, so it might actually be useful to have these update scripts in place for that time.

**Review by panicgh — APPROVED** [review]
I'm no expert in update scripts, but they look good to me. I tried the TBB one and it worked for me (updating both x86_64 and i686 hashes).

**Comment by SuperSandro2000** [comment]
Can you rebase? Otherwise LGTM and we can merge it afterwards.

**Comment by felschr** [comment]
I don't have access to my computer this week. I'll do it next week when I'm back home.

**Comment by felschr** [comment]
@SuperSandro2000 done

**Pull request merged**